### ISSUE TITLE
Interface "missing methods"

### ISSUE THREAD

**Issue by 131072** [gopherjs/gopherjs#808]
For some reason, when compiling the [Gorgonia library](https://github.com/gorgonia/gorgonia), GopherJS fails with:
```
../../../gorgonia.org/tensor/defaultengine_mapreduce.go:29:13: invalid operation: v (variable of type View) has no field or method Clone
../../../gorgonia.org/tensor/tensor.go:145:19: cannot use t.Slice(slices...) (value of type View) as Tensor value in assignment: missing method Clone
```
However, this error does not occur in standard Go, and I can only get it to work with GopherJS when I adjust the View interface from:
```go
// View is any Tensor that can provide a view on memory
type View interface {
	Tensor
	IsView() bool
	IsMaterializable() bool
	Materialize() Tensor
}
```
to replace the ```Tensor``` line with the entire definition of the Tensor interface:
```go
// View is any Tensor that can provide a view on memory
type View interface {
	// info about the ndarray
	Shape() Shape
	Strides() []int
	Dtype() Dtype
	Dims() int
	Size() int
	DataSize() int

	// Data access related
	RequiresIterator() bool
	Iterator() Iterator

	// ops
	Slicer
	At(...int) (interface{}, error)
	SetAt(v interface{}, coord ...int) error
	Reshape(...int) error
	T(axes ...int) error
	UT()
	Transpose() error // Transpose actually moves the data
	Apply(fn interface{}, opts ...FuncOpt) (Tensor, error)

	// data related interface
	Zeroer
	MemSetter
	Dataer
	Eq
	Cloner

	// type overloading methods
	IsScalar() bool
	ScalarValue() interface{}

	// engine/memory related stuff
	// all Tensors should be able to be expressed of as a slab of memory
	// Note: the size of each element can be acquired by T.Dtype().Size()
	Engine() Engine             // Engine can be nil
	MemSize() uintptr           // the size in memory
	Uintptr() uintptr           // the pointer to the first element, as a uintptr
	Pointer() unsafe.Pointer    // the pointer to the first elemment as a unsafe.Ponter
	IsNativelyAccessible() bool // Can Go access the memory
	IsManuallyManaged() bool    // Must Go manage the memory

	// formatters
	fmt.Formatter
	fmt.Stringer

	// all Tensors are serializable to these formats
	WriteNpy(io.Writer) error
	ReadNpy(io.Reader) error
	gob.GobEncoder
	gob.GobDecoder

	standardEngine() standardEngine
	headerer
	arrayer
	IsView() bool
	IsMaterializable() bool
	Materialize() Tensor
}
```

**Comment by myitcv** [comment]
This is actually an instance of https://github.com/golang/go/issues/25008 (GopherJS uses `go/types` under the covers). I've asked the question in that issue whether the fix could be back-ported to the 1.10 series. 

Thanks for raising in any case; I suggest we leave this open, depending on the response in the linked issue.

**Comment by 131072** [comment]
Thank you!

**Comment by myitcv** [comment]
@131072 per the linked issue, as things stand we are not going to get a fix for this until Go 1.11, and even then until we release a GopherJS corresponding to that version. 

So for now I've pushed up #811 in order to work around this as Robert suggested.

You can try this out locally:

```
cd $(go list -f '{{.Dir}}' github.com/gopherjs/gopherjs) && \
  git fetch origin pull/811/head && \
  git checkout FETCH_HEAD && \
  go install
```

With this version of `gopherjs`, `gorgonia.org/gorgonia` appears to compile just fine.

**Comment by 131072** [comment]
Alright, sounds good!
